hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/service/ContractCallService.java

```diff
@@ -45,27 +47,32 @@ public class ContractCallService {
 
     private final Counter.Builder gasCounter =
             Counter.builder("hedera.mirror.web3.call.gas").description("The amount of gas consumed by the EVM");
-    private final MirrorEvmTxProcessorFacade mirrorEvmTxProcessorFacade;
+    private final ApplicationContext ctx;
     private final MeterRegistry meterRegistry;
     private final BinaryGasEstimator binaryGasEstimator;
+    private final Store store;
 
     public String processCall(final CallServiceParameters params) {
         var stopwatch = Stopwatch.createStarted();
         var stringResult = "";
 
         try {
+            ThreadLocalHolder.startThread(store.getStackedStateFrames(), ctx);
```

Comment:
What about making `ContractCallContext` `AutoCloseable` and use try with resources? Then we could avoid the cleanThread call.

```suggestion
        try (var context = ContractCallContext.create(store.getStackedStateFrames(), mirrorEvmTxFactory)) {
```

Reply:
ContractCallContext is now AutoCloseable.